cdap-common/src/main/java/co/cask/cdap/common/guice/LocationRuntimeModule.java

```diff
@@ -80,6 +80,26 @@ private RootLocationFactory providesRootLocationFactory(CConfiguration cConf) {
     }
   }
 
+  public static UserGroupInformation getUserGroupInformation(CConfiguration cConf) {
```

Comment:
Actually see my "refactoring into Provider" comment below, then this method can simply be a private method inside the Provider class.